### PR TITLE
Add Nighthawk docs

### PR DESCRIPTION
**Pull request by navendu-pottekkat**
Signed-off-by: Navendu Pottekkat <navendupottekkat@gmail.com>

**Description**

This PR adds docs for using Nighthawk, Using GetNighthawk APIs and using Nighthawk with Meshery and SMP to run performance benchmarks.

- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by warunicorn19** [comment]
@amino19 label commenter workflow is failing.

**Comment by navendu-pottekkat** [comment]
Updated this page: https://deploy-preview-181--getnighthawk.netlify.app/docs

Will add more docs as we update Performance Management docs in Meshery

**Comment by leecalcote** [comment]
> @amino19 label commenter workflow is failing.

I went ahead and fixed this workflow.

File: docs/collections/_docs/building-nighthawk.md [hunk 1]
```diff
@@ -0,0 +1,398 @@
+---
+layout: docs
+title: Building and Using Nighthawk
+section: "Getting Started"
+---
+
+# Building Nighthawk
+
+Although GetNighthawk would provide all of the features available in Nighthawk, you can also build Nighthawk and run it from source.
```
Comment (leecalcote) [inline docs/collections/_docs/building-nighthawk.md:9]:
```suggestion
While GetNighthawk distribution are a convenient way of using Nighthawk, you can also build Nighthawk and run it from source.
```

File: docs/collections/_docs/building-nighthawk.md [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+Displays usage information and exits.
+
+<uri format>
+URI to benchmark. http:// and https:// are supported, but in case of
+https no certificates are validated. Provide a URI when you need to
+benchmark a single endpoint. For multiple endpoints, set
+--multi-target-* instead.
+
+
+L7 (HTTP/HTTPS/HTTP2) performance characterization tool.
```
Comment (leecalcote) [inline docs/collections/_docs/building-nighthawk.md:315]:
```suggestion
```

File: docs/collections/_docs/building-nighthawk.md [hunk 3]
```diff
@@ -0,0 +1,398 @@
...
+
+## Building Nighthawk
+
+Run:
+
+```
+bazel build -c opt //:nighthawk
+```
+
+# Using the Nighthawk CLI
```
Comment (leecalcote) [inline docs/collections/_docs/building-nighthawk.md:54]: Move this section out of the building-nighthawk page into a new page.

File: docs/collections/_docs/performance-benchmarking.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+layout: docs
+title: Running Performance Tests
+section: "Performance Benchmarking with Nighthawk"
+---
+
+[Meshery](https://meshery.io/) uses Nighthawk as one of its load generators to run performance benchmarking. Meshery is the canonical implementation of [Service Mesh Performance](https://smp-spec.io/), CNCF's service mesh performance benchmarking specification.
+
+# Performance Benchmarking Using mesheryctl
+
+mesheryctl is the command line interface of Meshery.
```
Comment (leecalcote) [inline docs/collections/_docs/performance-benchmarking.md:11]:
```suggestion
<pre>mesheryctl</pre> is the command line interface of Meshery. <pre>mesheryctl</pre> along with other load generators,  <pre>mesheryctl</pre> can use Nighthawk to run performance benchmarks.
```

File: docs/collections/_docs/performance-benchmarking.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+mesheryctl can use Nighthawk underneath the hood to run performance benchmarks.
```
Comment (leecalcote) [inline docs/collections/_docs/performance-benchmarking.md:13]:
```suggestion
```

File: docs/collections/_docs/performance-benchmarking.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+  rps: 10
+  headers: {}
+  cookies: {}
+  body: ""
+  content_type: ""
+  endpoint_urls:
+  - http://localhost:2323
+duration: "15s"
+```
+
```
Comment (leecalcote) [inline docs/collections/_docs/performance-benchmarking.md:79]: Need example of `mesheryctl perf apply -f` here.

**Comment by navendu-pottekkat** [comment]
@leecalcote Made the suggested changes. Good to merge?

**Review by leecalcote — APPROVED** [review]

**Pull request merged**